src/utils.ts

```diff
@@ -208,3 +221,54 @@ export const normalizePath = (filePath: string): string[] => {
     const absPath = path.resolve(filePath);
     return [path.basename(absPath), absPath];
 };
+
+export const getFileHash = (filename: string): string => {
+    const fileBuffer = fs.readFileSync(filename);
+    const hashSum = crypto.createHash('md5');
+    hashSum.update(fileBuffer);
+    return hashSum.digest('hex');
+};
+
+export const getCachedResults = (context: vscode.ExtensionContext, fileHash: string, logger: Logger): FileScanCacheEntry | undefined => {
+    logger.debug(`Getting cached results for hash ${fileHash}`);
+    validateCacheExpiration(context, logger);
+    const cache: ResultsCacheObject | undefined = context.workspaceState.get(cacheResultsKey);
+    return cache ? cache[fileHash] : undefined;
+};
+
+export const saveCachedResults = (context: vscode.ExtensionContext, fileHash: string, filename: string, results: FailedCheckovCheck[], logger: Logger): void => {
+    logger.debug(`Saving results for file ${filename} (hash: ${fileHash})`);
+    validateCacheExpiration(context, logger);
+
+    const cache: ResultsCacheObject | undefined = context.workspaceState.get(cacheResultsKey);
+    if (cache) {
+        cache[fileHash] = { fileHash, filename, results };
+    }
+};
+
+export const clearCache = (context: vscode.ExtensionContext, logger: Logger): void => {
+    logger.debug('Clearing results cache');
+    context.workspaceState.update(cacheResultsKey, undefined);  // undefined removes the key
+    context.workspaceState.update(cacheDateKey, undefined);
+};
+
+const getDate = (): number => {
+    const today = new Date();
+    today.setHours(0, 0, 0, 0);
```

Comment:
this will be like the end of day of `today`?

Reply:
It will reset every day at midnight, local time

Reply:
```
> t = new Date()
2022-05-01T14:30:45.189Z
> t.setHours(0,0,0,0);
1651352400000
> t
2022-04-30T21:00:00.000Z
```

The last line shows that the result is midnight, local time